### ISSUE TITLE
Other packages relying on `generics::tidy` can cause problems with `broom` <= 0.5

### ISSUE THREAD

**Issue by lukesonnet** [tidymodels/broom#528]
We maintain packages that have `tidy` methods. In anticipation of `broom` switching over to `generics` and with the hope of avoiding the dependencies `broom` has, we decided to import from `generics` once it went on CRAN.

However, if `broom <= 0.5.0` is loaded after `estimatr >= 0.14.0`, `broom::tidy` masks `tidy` from `estimatr` and our tidy methods don't work. This problem is solved with the GitHub version of `broom` but not the version currently on CRAN.

Here's a MWE:
```r
install.packages("broom")
install.packages("estimatr")

library(estimatr)
library(broom)

# errors!
tidy(lm_robust(mpg ~ hp, mtcars))
```

This can be solved by our users upgrading `broom` from GitHub or loading `estimatr` second. However, the preferred solution would be to have the new version of `broom` on CRAN. Please let us know if there is a timeline for submission and if there's anything we can do to help.

**Comment by alexpghayes** [comment]
@topepo the revdep check on this is going to be a bit of a pain since I no longer have the RStudio MBP. Can I borrow some cores from you later this week? Will create a branch with the update in a day or two. Also, any thoughts on how to publicize this? Should we go a general blogpost on all the tidymodels packages upgrading to `generics` in the next couple days?

**Comment by lukesonnet** [comment]
@alexpghayes , if it is simple to set up the testing environment and all you need is computing power, DeclareDesign could cover those costs.

**Comment by topepo** [comment]
I can run it if you need (since I appear to be the designated revdep compute center 😜). 

If you think that it is a big update with a lot of changes, then let's make it its own blog post. Otherwise, just send me a few paragraphs and we'll bundle it in to the upcoming blog post.

**Comment by alexpghayes** [comment]
A patch to use update to generics is in progress on the 0.5.1 branch (https://github.com/tidymodels/broom/tree/0.5.1). Once Jenny gets #534 or whatever that becomes merged would love to get a round of revdep results, then will send to CRAN!

**Comment by strengejacke** [comment]
I have a question using the generics-package, also posted here:
https://github.com/r-lib/generics/issues/37

Short: Can I replace my imports of broom with generics? If so, and my package then depends on generics instead of broom, how will users get broom or broom.mixed installed along with my package, when my package no longer depends on broom/broom.mixed?

**Comment by alexpghayes** [comment]
The TL;DR is that you should import `broom` if you need any functionality present in the `broom` (i.e. you use `tidy()`, `glance()` or `augment()` in your package). You should *re-export* `tidy()`, `glance()` and `augment()` if your package provides *new* S3 methods for objects not tidied by `broom`.

Based on the linked issue, it looks like you're using `broom` rather than implementing new S3 methods, so you should definitely import `broom`. No need for `generics`. `generics` provides generic functions, but no methods (in your case it looks like you need the methods themselves). Some details on the differences between S3 generics and S3 methods can be found at https://adv-r.hadley.nz/s3.html.

I'll try to improve the documentation on this over winter break.

**Comment by strengejacke** [comment]
Ok, thanks for this clarification!
